[Backport stable/8.8] feat: introduce parameter for returning full variable values

backport-action opened the pull request:
# Description
Backport of #40884 to `stable/8.8`.

relates to camunda/camunda#39189

kristinkomschow committed:
test: added test coverage for includeFullValue search parameter

(cherry picked from commit 7884945ad9e9dd475ba2261de5790bd281d468c7)

kristinkomschow committed:
test: refactored user task and variable search tests

- using TestHelper methods instead of private test methods
- applying IDE suggestions to simplify `isEqualTo` assertions, joining assertions, removing unused variables

(cherry picked from commit 080ee2a9596476bb1f4d1ba65b23ce2dbc5f711c)

tmetzke commented:
I guess this mostly fails due to the split-up API spec files in 8.9. If you need any assistance here, @kristinkomschow, don't hesitate. Happy to help if needed.

kristinkomschow committed:
feat: introduce parameter for returning full variable values

relates to https://github.com/camunda/camunda/issues/39189

(cherry picked from commit 0abf1f37fa7411af70b5241ac7e15a4f432eb2af)

# Conflicts:
#	zeebe/gateway-protocol/src/main/proto/v2/user-tasks.yaml
#	zeebe/gateway-protocol/src/main/proto/v2/variables.yaml

tmetzke reviewed (approved):
👍 Looks good to me, let's see what the CI says 🙂

kristinkomschow commented:
@nicpuppa while this is technically a feature, we are introducing it to fix a bug in Tasklist that exists in 8.8 and could only be temporarily patched with an error message when it was discovered while the release process was ongoing.

The full discussion about fixing this is in the slack channel here: https://camunda.slack.com/archives/C09K31PGFEG/p1763052311295529
